src/quo/components/avatars/user_avatar/view.cljs

```diff
@@ -26,6 +29,59 @@
        :weight :semi-bold}
       (utils.string/get-initials full-name amount-initials)]]))
 
+(defn icon-avatar
+  [{:keys [size customization-color]
+    :or   {customization-color :blue}}]
+  (let [theme (quo.theme/use-theme)]
+    [rn/view
+     {:accessibility-label :icon-avatar
+      :style               (style/initials-avatar size customization-color theme)}
+     [icon/icon (if (= size :big) :i/user :i/friend)
+      {:size  (style/default-user-icon-size size)
+       :color colors/white-opa-70}]]))
+
+(defn- profile-picture-source
+  [{:keys [picture-config profile-picture size full-name status-indicator? online? theme
+           outer-styles ring?]}]
+  (let [amount-initials (if (#{:xs :xxs :xxxs} size) 1 2)
+        font-size       (get-in style/sizes [size :font-size])
+        sizes           (get style/sizes size)
+        indicator-color (get (style/indicator-color theme) (if online? :online :offline))
+        font-size       (:font-size (text/text-style {:size font-size} nil))]
+    (cond
+      picture-config
+      {:uri (image-server/get-image-uri
+             picture-config
+             {:length                   amount-initials
+              :full-name                full-name
+              :font-size                font-size
+              :indicator-size           (when status-indicator?
+                                          (:status-indicator sizes))
+              :indicator-border         (when status-indicator?
+                                          (:status-indicator-border sizes))
+              :indicator-center-to-edge (when status-indicator?
+                                          (:status-indicator-center-to-edge sizes))
+              :indicator-color          indicator-color
+              :theme                    theme
+              :color                    (:color style/initials-avatar-text)
+              :size                     (:width outer-styles)
+              :ring?                    ring?
+              :ring-width               (:ring-width sizes)})}
+
+      (:uri profile-picture)
+      profile-picture
+
+      (number? profile-picture)
+      profile-picture
+
+      :else {:uri profile-picture})))
```

Comment:
Refactor: just extracted this from the function below, the logic is complex enough to have its own function